src/mempool.rs

```diff
@@ -217,20 +234,38 @@ impl FeeHistogram {
         Self::new(std::iter::empty())
     }
 
+    fn bin_index(fee: Amount, vsize: u64) -> usize {
+        let fee_rate = fee.to_sat() / vsize;
+        usize::try_from(fee_rate.leading_zeros()).unwrap()
+    }
+
     fn new(items: impl Iterator<Item = (Amount, u64)>) -> Self {
-        let mut result = FeeHistogram::default();
+        let mut histogram = FeeHistogram::default();
         for (fee, vsize) in items {
-            let fee_rate = fee.to_sat() / vsize;
-            let index = usize::try_from(fee_rate.leading_zeros()).unwrap();
-            // skip transactions with too low fee rate (<1 sat/vB)
-            if let Some(bin) = result.vsize.get_mut(index) {
-                *bin += vsize
-            }
-            if let Some(bin) = result.count.get_mut(index) {
-                *bin += 1
-            }
+            let bin_index = FeeHistogram::bin_index(fee, vsize);
+            histogram.insert(bin_index, vsize);
+        }
+        histogram
+    }
+
+    fn insert(&mut self, bin_index: usize, vsize: u64) {
+        // skip transactions with too low fee rate (<1 sat/vB)
+        if let Some(bin) = self.vsize.get_mut(bin_index) {
+            *bin += vsize
+        }
+        if let Some(bin) = self.count.get_mut(bin_index) {
+            *bin += 1
+        }
+    }
+
+    fn remove(&mut self, bin_index: usize, vsize: u64) {
+        // skip transactions with too low fee rate (<1 sat/vB)
+        if let Some(bin) = self.vsize.get_mut(bin_index) {
+            *bin = bin.saturating_sub(vsize);
```

Comment:
Maybe it's safer to use `bin.checked_sub(vsize).expect(...)`? 
(since we shouldn't have negative values here - unless there is a bug)

Reply:
Sure, and probably log a `warn!` call if we somehow hit a `checked_sub` error. Fixed by amending dd8f994